src/admin/mod.rs

```diff
@@ -121,9 +123,13 @@ impl Server {
                 }
             }));
 
+        let shutdown_trigger = shutdown.trigger();
         tokio::spawn(async move {
             info!("Serving admin server at {}", self.addr);
-            server.await
+            if let Err(err) = server.await {
```

Comment:
So this seems to wait for the server to terminate then trigger shutdown. But the server should never terminate.

I think the intent here was to offer a /quitquitquit endpoint like Envoy that explicitly calls `shutdown_trigger.shutdown_now()`

Reply:
Hi @howardjohn, okay, got it. will add an new endpoint for admin server's shutdown. And I think that the error catch for admin server is necessary.  Does it make sense?

Reply:
Hi @howardjohn, an new endpoint `quitquitquit` has been added.

Reply:
Thanks! Do you know what case would actually trigger server.await to return an error btw? It seemed non-obvious to me.

Reply:
Hi @howardjohn, u r welcome, I think this kind of server will always occur some accidents especially when the IP and port are known by all. So I think it's necessary to catch the server internal error to analyze.